Doc 1497/improve in app notifications and integrate sca guide (#596)
swan-io/docs: docs/topics/users/consent/guide-configure-notifications.mdx

You can declare a preferred notification channel for your users and configure how your consent notifications appear.

If you're using **Swan's Web Banking interface**, Swan sends a text message to your users for one-time passwords, logins, and consents, regardless of preferred channel and configuration.

:::info Timeout
Your integration has two seconds to respond to a consent notification.
If your response is anything other than a `200 OK HTTP`, or there's no response at all, Swan redirects the user to the standard text message consent flow.
This timeout complies with 3-D Secure (3DS) requirements.
:::

## When to use consent notifications {#when-to-use}

Consent notifications are relevant when you have both desktop and mobile apps, or when you want to optimize 3-D Secure (3DS) on mobile.

:::info Consent flows only
Consent notifications are only available for Consent flows (sensitive operations). They can't be used for OAuth flows such as login or authentication.
:::

## How consent notifications work {#how-it-works}

> By default, when a user performs a sensitive operation on desktop and needs to consent, Swan sends an SMS to their smartphone with the consent link.

Reviewer: ```suggestion
By default, when a user performs a sensitive operation on desktop and needs to consent, Swan sends a text message to their smartphone with the consent link.
```

With consent notifications configured, when a user performs a sensitive operation on desktop, your mobile app receives a notification so the user can consent inside your mobile app. This redirects desktop users to your mobile app to give consent.

> Think of consent notifications as another way to give consent alongside SMS and QR codes. This is similar to how some purchases on desktop sometimes require approval from your banking app: you initiate an action on desktop and approve it on your mobile app.

Reviewer: ```suggestion
Think of consent notifications as another way to give consent alongside text messages and QR codes. This is similar to how some purchases on desktop sometimes require approval from your banking app: you initiate an action on desktop and approve it on your mobile app.
```

Along with a notification, you'll receive an `authorizationUrl` and the `consent` object to display to your user on their mobile.
On desktop, Swan displays the `consentUrl`.

While consent is ongoing, the user's mobile plays a repeated image with your accent color and logo.

## Tips & recommendations {#tips}

> - If Swan detects the user is already on a mobile app, Swan uses a mobile consent flow and doesn't send an SMS. This is different from consent notifications, which enable communication between browser and mobile app for in-app notification consent instead of SMS.

Reviewer: ```suggestion
- If Swan detects that the user is already on a mobile app, Swan uses a mobile consent flow and doesn't send a text message. This differs from consent notifications, which enable communication between the browser and the mobile app for in-app notification consent instead of text message consent.
```

Reviewer: Desktop browser? Or any browser?

> - 3DS always uses SMS regardless of whether you're on desktop or mobile. However, you can use consent notifications to optimize 3DS UX by setting `preferredNotificationChannel: App`. When configured, users performing 3DS card payment validation receive an in-app notification instead of an SMS, containing a consent link that opens the consent app directly on their phone.

Reviewer: Replace all instances of "SMS" with "text message".
